elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/QueryBuilders.java

```diff
@@ -0,0 +1,425 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.adapter.elasticsearch;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * Utility class to generate elastic search queries. Most query builders have
+ * been copied from ES distribution. The reason we have separate definition is
+ * high-level client dependency on core modules (like lucene, netty, XContent etc.) which
+ * is not compatible between different major versions.
+ *
+ * <p>Jackson API is used to generate ES query as JSON document.
+ */
+class QueryBuilders {
+
+  private QueryBuilders() {
+    // util class
```

Comment:
This comment is not necessary and the constructor can be written on one line with empty braces. This convention is already used quite a lot.